modules/ximgproc/src/brightedges.cpp

```diff
@@ -0,0 +1,233 @@
+
+		line =
+			input[iedge.cols * (row - 2) + (col)] +
+			input[iedge.cols * (row - 1) + (col)] +
+			input[iedge.cols * (row + 1) + (col + 1)] +
+			input[iedge.cols * (row + 2) + (col + 1)];
+		if (line == 0) lines += 1;
+
+		line =
+			input[iedge.cols * (row - 2) + (col - 1)] +
+			input[iedge.cols * (row - 1) + (col - 1)] +
+			input[iedge.cols * (row + 1) + (col)] +
+			input[iedge.cols * (row + 2) + (col)];
+		if (line == 0) lines += 1;
+
+		line =
+			input[iedge.cols * (row - 2) + (col - 2)] +
+			input[iedge.cols * (row - 1) + (col - 2)] +
+			input[iedge.cols * (row - 2) + (col - 1)] +
+			input[iedge.cols * (row - 1) + (col - 1)] +
+			input[iedge.cols * (row + 1) + (col + 1)] +
+			input[iedge.cols * (row + 1) + (col + 2)] +
+			input[iedge.cols * (row + 2) + (col + 1)] +
+			input[iedge.cols * (row + 2) + (col + 2)];
+		if (line < weight) lines += 1;
+
+		// Near vertical 
+		line =
+			input[iedge.cols * (row - 2) + (col)] +
+			input[iedge.cols * (row - 1) + (col)] +
+			input[iedge.cols * (row - 2) + (col - 1)] +
+			input[iedge.cols * (row - 2) + (col + 1)] +
+			input[iedge.cols * (row + 1) + (col)] +
+			input[iedge.cols * (row + 2) + (col)] +
+			input[iedge.cols * (row + 2) + (col + 1)] +
+			input[iedge.cols * (row + 2) + (col - 1)];
+		if (line < weight) lines += 1;
+		// Near diagonal top right to bottom left 
+
+		line =
+			input[iedge.cols * (row - 2) + (col + 2)] +
+			input[iedge.cols * (row - 1) + (col + 1)] +
+			input[iedge.cols * (row - 2) + (col - 1)] +
+			input[iedge.cols * (row - 1) + (col + 2)] +
+			input[iedge.cols * (row + 1) + (col - 1)] +
+			input[iedge.cols * (row + 2) + (col - 2)] +
+			input[iedge.cols * (row + 2) + (col - 1)] +
+			input[iedge.cols * (row + 1) + (col - 2)];
+		if (line < weight) lines += 1;
+
+		// Near horizontal 
+		line =
+			input[iedge.cols * (row)+(col - 2)] +
+			input[iedge.cols * (row)+(col - 1)] +
+			input[iedge.cols * (row - 1) + (col - 2)] +
+			input[iedge.cols * (row + 1) + (col - 2)] +
+			input[iedge.cols * (row)+(col + 1)] +
+			input[iedge.cols * (row)+(col + 2)] +
+			input[iedge.cols * (row + 1) + (col + 2)] +
+			input[iedge.cols * (row - 1) + (col + 2)];
+		if (line < weight) lines += 1;
+
+		if (line == 1) return 0;
+
+		int surround = input[iedge.cols * (row - 1) + (col - 1)] +
+			input[iedge.cols * (row - 1) + (col)] +
+			input[iedge.cols * (row - 1) + (col + 1)] +
+			input[iedge.cols * (row)+(col - 1)] +
+			input[iedge.cols * (row)+(col + 1)] +
+			input[iedge.cols * (row + 1) + (col - 1)] +
+			input[iedge.cols * (row + 1) + (col)] +
+			input[iedge.cols * (row - 1) + (col + 1)];
+
+		if (surround == 8 * 255) return 255;
+		if (surround == 0) return 255;
+
+		return input[iedge.cols * (row)+(col)];
+
+
+	}
+	
+	int contrastEdges(Mat &minput, Mat &mouput, int contrast) {
+
+		unsigned char *input = (unsigned char*)(mouput.data);
+
+		Mat mwork = minput.clone();
+		mwork.setTo(Scalar(255));
+
+		mouput.setTo(Scalar(255));
```

Comment:
This assignment could be done later when `mwork` is already filled to allow inplace processing as well.

Reply:
@terfendail Corrected the code to only explicitly work on the border instead of setting all matrix to 255. Also corrected the minimum detection. (FYI in my project we don't use contrasting as convolutions do the job for us). See the new sample image (I tested on many different images).
![testbrightedges](https://user-images.githubusercontent.com/5433466/32687508-124bce24-c6be-11e7-883a-a1c281eb38a3.png)